parser-typechecker/src/Unison/PatternMatchCoverage/Solve.hs

```diff
@@ -518,12 +518,9 @@ addConstraint con0 nc = do
     C.PosLit var pmlit ->
       let updateLiteral pos neg lit
             | Just lit1 <- pos,
-              lit1 == lit = case lit1 == lit of
-                -- we already have this positive constraint
-                True -> (pure (), Ignore)
-                -- contradicts positive info
-                False -> (contradiction, Ignore)
-            -- the constraint contradicts negative info
+              lit1 == lit =
+                (pure (), Ignore) -- we already have this positive constraint
+                -- the constraint contradicts negative info
```

Comment:
This was some weird dead code? But we can delete the comment for the deleted case.
```suggestion
              lit1 == lit =
                 -- we already have this positive constraint
                (pure (), Ignore)
```

Reply:
Yeah GHC has been steadily getting better at detecting unreachable cases 😎

Reply:
@aryairani I think that comment (`-- the constraint contradicts negative info`) was actually for the other case `Set.member lit neg = (contradiction, Ignore)`.  Initially the code looked like this:
```
            | Just lit1 <- pos,
              lit1 == lit = case lit1 == lit of
                -- we already have this positive constraint
                True -> (pure (), Ignore)
                -- contradicts positive info
                False -> (contradiction, Ignore)
            -- the constraint contradicts negative info
            | Set.member lit neg = (contradiction, Ignore)
```

After removing the `False` case should we get:
```
            | Just lit1 <- pos,
              lit1 == lit =
                -- we already have this positive constraint
                True -> (pure (), Ignore)
            -- the constraint contradicts negative info
            | Set.member lit neg = (contradiction, Ignore)
```
I think ormolu though got slightly confused here and indented the comment.

WDYT?  I'm happy to either re-add it or if you reckon it's clear enough we can just leave it deleted.

Reply:
Oh... err... yeah that was a mistake. If you are willing to re-add it, that would be great.